compose-builder/tui-generator.sh

```diff
@@ -0,0 +1,299 @@
+####################################################################
+function displayNoSecty() {
+    message="Would you like to generate a non-secure configuration of the Compose file?"
+    $WHIPTAIL --title "Non-Secure Configuration" --yesno "$message" $LINES $COLUMNS
+}
+
+
+####################################################################
+#    Device Services Options Display Function                      #
+####################################################################
+function devServiceOption() {
+    message="Available Device Services to include in your compose file.\n
+            Press <SPACEBAR> to Select \n
+            Press <ENTER> to Skip
+            "
+
+    # Generate a Specific form of Array required by whiptail checklist
+    # FORMAT: "<INDEX> <DESCRIPTION> <OFF>"
+    arglist=()
+    for index in "${deviceServices[@]}";
+    do
+        arglist+=("$index")
+        arglist+=("${deviceServiceDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_DEVSERVICES=$($WHIPTAIL --title "Device Services" \
+                --ok-button Next \
+                --nocancel \
+                --notags --separate-output \
+                --checklist "$message" $LINES $COLUMNS $(( $LINES - 12 )) \
+                -- "${arglist[@]})" \
+                3>&1 1>&2 2>&3)
+}
+
+####################################################################
+#    Message Bus Options Display Function                          #
+####################################################################
+function msgBusOption() {
+    message="Available Message Buses to replace the default one.\n
+            Press <SPACEBAR> to Select \n
+            Press <ENTER> to Skip
+            "
+
+    # Generate a Specific form of Array required by whiptail checklist
+    # FORMAT: "<INDEX> <DESCRIPTION> <OFF>"
+    arglist=()
+    for index in "${messageBus[@]}";
+    do
+        arglist+=("$index")
+        arglist+=("${msgBusDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_BUS=$($WHIPTAIL --title "App Services" \
+                --ok-button Next \
+                --nocancel \
+                --notags --separate-output \
+                --radiolist "$message" $LINES $COLUMNS $(( $LINES - 12 )) \
+                -- "${arglist[@]})" \
+                3>&1 1>&2 2>&3)
+}
+
+####################################################################
+#    Generate / Generate & Run Option Display Function             #
+####################################################################
+function finalStep() {
+    message="What would you like to do?"
+    $WHIPTAIL --title "Non-Secure Configuration" \
+        --yes-button "Generate File" \
+        --no-button "Generate File and Run" \
+         --yesno "$message" $LINES $COLUMNS
+}
+
+## Step - 1: Start by Asking about whether Images should be pulled for ARM64?
+displayArm64
+if [[ $? == 0 ]]; then
```

Comment:
check the result of the call directly:
```suggestion
if displayArm64; then
```